src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -0,0 +1,297 @@
+  unclaimedRewards: string
+} & EarnOpportunityType
+
+export const FoxEthProvider = ({ children }: FoxEthProviderProps): JSX.Element => {
+  const {
+    state: { wallet },
+  } = useWallet()
+  const ethAsset = useAppSelector(state => selectAssetById(state, ethAssetId))
+
+  const chainAdapterManager = getChainAdapterManager()
+  const adapter = chainAdapterManager.get(ethAsset.chainId) as ChainAdapter<KnownChainIds>
+
+  const [lpLoading, setLpLoading] = useState<boolean>(true)
+  const [lpFoxBalance, setLpFoxBalance] = useState<string | null>(null)
+  const [lpEthBalance, setLpEthBalance] = useState<string | null>(null)
+  const [ongoingTxId, setOngoingTxId] = useState<string | null>(null)
+  const [foxEthLpOpportunity, setFoxEthLpOpportunity] = useState<EarnOpportunityType>(lpOpportunity)
+  const { calculateHoldings, getLpTVL } = useFoxEthLiquidityPool()
+
+  const [connectedWalletEthAddress, setConnectedWalletEthAddress] = useState<string | null>(null)
+  const [farmingLoading, setFarmingLoading] = useState<boolean>(true)
+  const [foxFarmingTotalBalance, setFoxFarmingTotalBalance] = useState<string>('')
+  const [foxFarmingOpportunities, setFoxFarmingOpportunities] = useState<
+    FoxFarmingEarnOpportunityType[]
+  >([v4FarmingOpportunity])
+  const { isLpAprLoaded, lpApr } = useLpApr()
+  const ethMarketData = useAppSelector(state => selectMarketDataById(state, ethAssetId))
+  const foxMarketData = useAppSelector(state => selectMarketDataById(state, foxAssetId))
+  const foxAssetPrecision = useAppSelector(state => selectAssetById(state, foxAssetId)).precision
+  const lpAssetPrecision = useAppSelector(state =>
+    selectAssetById(state, foxEthLpAssetId),
+  ).precision
+
+  useEffect(() => {
+    if (wallet && adapter) {
+      ;(async () => {
+        const address = await adapter.getAddress({ wallet })
+        setConnectedWalletEthAddress(address)
+      })()
+    }
+  }, [adapter, wallet])
+
+  const reloadFarmingOpportunities = useCallback(async () => {
+    if (!connectedWalletEthAddress) return
+    try {
+      const newOpportunities = await Promise.all(
+        foxFarmingOpportunities.map(async opportunity => {
+          const data = await getOpportunityData({
+            contractAddress: opportunity.contractAddress,
+            ethAssetPrecision: ethAsset.precision,
+            ethPrice: ethMarketData.price,
+            lpAssetPrecision,
+            address: connectedWalletEthAddress,
+            foxPrice: foxMarketData.price,
+            foxAssetPrecision,
+          })
+          if (!data) return opportunity
+          const { tvl, apr, balances, expired } = data
+          return {
+            ...opportunity,
+            cryptoAmount: balances.cryptoBalance,
+            fiatAmount: balances.fiatBalance,
+            isLoaded: isLpAprLoaded,
+            unclaimedRewards: balances.unclaimedRewards,
+            expired,
+            apy: lpApr
+              ? bnOrZero(apr)
+                  .plus(lpApr ?? 0)
```

Comment:
You're checking for `lpApr` truthiness two lines above, this will never evaluate to the right side
```suggestion
                  .plus(lpApr)
```